File: src/primes.rs
fn is_prime(number: usize) -> bool {
    if number == 2 || number == 3 {
        return true;
    }

    if number <= 1 || number % 2 == 0 || number % 3 == 0 {
        return false;
    }

    for i in (5..).step_by(6) {
        if i * i > number {
            break;
        }
        if number % i == 0 || number % (i + 2) == 0 {
            return false;
        }
    }

    true
}

pub(crate) fn primes_add_up_to_number(number_to_check: usize) -> bool {
    if is_prime(number_to_check - 2) {
        return true;
    }

    (3..)
        .step_by(2)
        .take((number_to_check - 3) / 2)
        .filter(|&number| is_prime(number))
        .any(|number: usize| is_prime(number_to_check - number))
}

File: src/Primes.java
import java.util.stream.IntStream;

public class Primes {
    private static boolean isPrime(int number) {
        if (number == 2 || number == 3) {
            return true;
        }

        if (number <= 1 || number % 2 == 0 || number % 3 == 0) {
            return false;
        }

        for (int i = 5; i * i <= number; i += 6) {
            if (number % i == 0 || number % (i + 2) == 0) {
                return false;
            }
        }

        return true;
    }

    public static boolean primesAddUpToNumber(int numberToCheck) {
        if (isPrime(numberToCheck - 2)) {
            return true;
        }

        return IntStream.iterate(3, i -> i + 2)
                .limit((numberToCheck - 3) / 2)
                .filter(Primes::isPrime)
                .anyMatch(i -> isPrime(numberToCheck - i));
    }
}
